W-14677283 update spec duke (#275)
mulesoft/docs-code-builder: modules/ROOT/pages/imp-implement-apis.adoc

This interface includes flows for each endpoint, a built-in XML-based router, and error handlers,
which you then implement within a Mule application.

The process you use to implement your API depends on whether you already

> created an integration or implementation project, or published the spec to Exchange:

Author: ```suggestion
created a project for your integration or implementation, or published the spec to Exchange:
```

* To scaffold an API using a spec from Exchange into:

** A new project, see <<scaffold-new-integration>>.
** An existing project, see <<import-spec-into-project>>.

* To update your project with the latest changes to a spec in Exchange, see <<re-scaffold-spec>>.

* To scaffold and re-scaffold an API into a project without publishing 
the spec to Exchange, see xref:imp-implement-local-apis.adoc[].

== Before You Begin

* xref:start-acb.adoc[Set up and access the web or desktop IDE].
* xref:des-create-api-specs.adoc[Create and test your API specification].
* xref:des-publish-api-spec-to-exchange.adoc[Publish your API spec to Exchange].
* Optionally, xref:int-create-integrations.adoc[create your integration project (Mule app)].

[[scaffold-new-integration]]
== Scaffold an API into a New Project

> If you have not already created an implementation project, use Anypoint Code Builder to scaffold the API in a new Mule project:

Author: ```suggestion
If you have not already created a project, use Anypoint Code Builder to scaffold the API in a new Mule project:
```

| *Show filters* |See xref:filter-search-results.adoc[] for more information.
|===

. Search for an API spec on Exchange:
+

> .. Start typing the name of an API specification.

Reviewer: should we always use spec instead of specification?

Author: We've started using "spec" these days ¯\_(ツ)_/¯ It's shorter and well understood.

Reviewer: exactly, wondering if also affects `Start typing the name of an API specification.`

Author: ohhh, thx for catching that

Author: conformed to new spelling everywhere.

</dependency>
--
+

> ** Scaffolds your API into the new implementation project and opens the configuration XML file.

Author: ```suggestion
** Scaffolds your API into the new Mule project and opens the configuration XML file.
```

+

> The configuration XML file includes the interface for your implementation project,

Author: ```suggestion
The configuration XML file includes the interface for your project,
```

You can now implement the interface within the Mule app.

[[re-scaffold-spec]]
== Re-scaffold an API Specification into a Project 

Re-scaffolding enables you to update a project in the IDE with the latest changes and version of an API spec in Anypoint Exchange. Re-scaffolding updates the configuration XML and `pom.xml` in your project. To re-scaffold a spec defined in a local API design project that is not published to Exchange, see xref:imp-implement-local-apis.adoc[], instead.

This procedure assumes that an updated version of the spec is available from Exchange, for example, a 2.0.0 version:

image::imp-assets-from-exchange-multiversion.png["Selecting a new version of an API spec from Exchange"]

> . In Anypoint Code Builder, open your implementation project in the Explorer view, for example, `my-implementation-project`.

Author: ```suggestion
. In Anypoint Code Builder, open your project in the Explorer view, for example, `my-implementation-project`.
```

// Pointer to Command Palette
include::partial$acb-reusable-steps.adoc[tags="open-command-palette"]
// Pointer to Import Asset from Exchange command
include::partial$acb-reusable-steps.adoc[tags="import-api-exchange"]
+
image::imp-assets-from-exchange-multiversion.png["Assets from Exchange dropdown"]
. At the prompt, select the version of the API spec, such as `v2.0.0`.
. When prompted to scaffold the API dependency, select *Yes*.
+
When you scaffold the API dependency, Anypoint Code Builder:

** Adds the API spec as a dependency in your project's `pom.xml` file:
+
[source,XML]
--
<dependency>
    <groupId>e91cab06-650b-4634-9c6f-5bc4f4fc4d17</groupId>
    <artifactId>phoneSync1005</artifactId>
    <version>2.0.0</version>
    <classifier>oas</classifier>
    <type>zip</type>
</dependency>
--

> ** Updates the configuration XML file that contains the interface for your implementation project with the changes to the imported specification.

Author: ```suggestion
** Updates the configuration XML file that contains the interface for your project with the changes to the imported specification.
```

> . Check for the changes to your implementation project.

Author: ```suggestion
. Check for the changes to your project.
```

+

> For example, if the updated specification adds a new endpoint, the configuration XML for the implementation project has a new flow for the endpoint. 

Author: ```suggestion
For example, if the updated specification adds a new endpoint, the configuration XML for the project has a new flow for the endpoint. 
```
